src/plugins/intel_gpu/src/plugin/ops/matmul.cpp

```diff
@@ -157,9 +157,32 @@ static void CreateMatMulOp(Program& p, const std::shared_ptr<ngraph::op::v0::Mat
 
         auto lastLayerName = layerName;
         if (reshape_fc) {
-            auto outputShape = tensor_from_dims(op->get_output_shape(0));
             auto outReshapeName = layerName + "_cldnn_out_reshape";
-            auto outReshapePrim = cldnn::reshape(outReshapeName, layerName, outputShape, op->get_friendly_name());
+
+            // add reorder
```

Comment:
And how about applying this change only when outDims.size() > 4 ?

Reply:
1. To reduce the effort of such reduction as many as possible 
2. To show the intention of adding reorder more clearly here